asset_proof.js

```diff
@@ -1,19 +1,26 @@
+const {TextEncoder} = require('./polyfill/text_encoder');
+const {btoa} = require('./polyfill/btoa');
+
 /**
  * @public
  */
 class AssetProof {
   /**
    * @param {string} id
    * @param {number} age
-   * @param {Uint8Array} hash
    * @param {string} nonce
+   * @param {string} input
+   * @param {Uint8Array} hash
+   * @param {Uint8Array} prevHash
    * @param {Uint8Array} signature
    */
-  constructor(id, age, hash, nonce, signature) {
+  constructor(id, age, nonce, input, hash, prevHash, signature) {
```

Comment:
we don't need to worry about the API change here because the upper layers (node and web SDKs) don't really construct AssetProof by themself.